_projects/nested-languages.md

```diff
@@ -24,16 +24,62 @@ tags:
   - Core
   - Bears
 ---
-Multiple programming languages can coexist in a single source file. coala would
+Generally, a single source file contains code for exactly one programming language.
+But multiple programming languages can also coexist in a single source file. coala would
 benefit from the ability to determine the programming language of each part of
 a file. If we have e.g. a PHP file, it is likely that we'd have chunks of html
 inside a few of them. We would like coala to be able to still run meaningful
 analysis on those files.
 Other examples include:
 
-* awk inside bash
-* JSON inside Javascript
-* HTML, LaTeX or others inside Jinja2
+* JSON inside Javascript is valid in some cases, for example JSONFormatBear for coala.
+* Web templating - languages such as PHP or JSP files mix code into HTML.
+* Cython, it is a superset of the Python language that additionally supports
+  calling C functions and declaring C types on variables and class attributes.
+* We can call Python code from Java using Jython, which is a Python implementation in Java.
+* reStructuredText and Markdown language specify different language for codeblock.
+
+  For example
+
+````
+```python
+s = "Python syntax highlighting"
+print s
+```
+````
+OUTPUT
+
+```python
+s = "Python syntax highlighting"
+print s
+```
+
+In the above example we have a code block which is embedded into markdown and
+language used to write a code block is python.
+
+
+
+
+```python
```

Comment:
here you start a new concept, with an example, and the add some thoughts below it.

People like to read a bit first, before seeing the example.

Also look at the rendered version of this and ask yourself if you would submit this document to a lecturer as a proposal for a project.